stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/message/internal/MessageSyncContentMapper.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.offline.repository.domain.message.internal
+
+import io.getstream.chat.android.client.models.MessageAwaitingAttachments
+import io.getstream.chat.android.client.models.MessageModerationFailed
+import io.getstream.chat.android.client.models.MessageSyncContent
+
+internal fun MessageSyncContentEntity.toModel(): MessageSyncContent {
+    return when (this) {
+        is MessageModerationFailedEntity -> MessageModerationFailed(
+            violations = violations.map { violation ->
```

Comment:
Similar thing here - do we need to map the same thing to itself? I imagine it's to preserve/decouple Entity stuff to domain level models

Reply:
We're mapping here:
- `MessageSyncContentEntity` into `MessageSyncContent`
- `ViolationEntity` into `Violation`